src/config.js

```diff
@@ -0,0 +1,1602 @@
+      } catch (e) {
+        return false
+      }
+    }
+
+    if (scopeSelector != null) {
+      this.setRawScopedValue(keyPath, value, source, scopeSelector)
+    } else {
+      this.setRawValue(keyPath, value)
+    }
+
+    if ((source === this.getUserConfigPath()) && shouldSave && !this.configFileHasErrors && this.settingsLoaded) {
+      this.requestSave()
+    }
+    return true
+  }
+
+  // Essential: Restore the setting at `keyPath` to its default value.
+  //
+  // * `keyPath` The {String} name of the key.
+  // * `options` (optional) {Object}
+  //   * `scopeSelector` (optional) {String}. See {::set}
+  //   * `source` (optional) {String}. See {::set}
+  unset (keyPath, options) {
+    if (!this.settingsLoaded) {
+      this.pendingOperations.push(() => this.unset(keyPath, options))
+    }
+
+    let {scopeSelector, source} = options != null ? options : {}
+    if (source == null) { source = this.getUserConfigPath() }
+
+    if (scopeSelector != null) {
+      if (keyPath != null) {
+        let settings = this.scopedSettingsStore.propertiesForSourceAndSelector(source, scopeSelector)
+        if (getValueAtKeyPath(settings, keyPath) != null) {
+          this.scopedSettingsStore.removePropertiesForSourceAndSelector(source, scopeSelector)
+          setValueAtKeyPath(settings, keyPath, undefined)
+          settings = withoutEmptyObjects(settings)
+          if (settings != null) { this.set(null, settings, {scopeSelector, source, priority: this.priorityForSource(source)}) }
+          if ((source === this.getUserConfigPath()) && !this.configFileHasErrors && this.settingsLoaded) {
+            return this.requestSave()
+          }
+        }
+      } else {
+        this.scopedSettingsStore.removePropertiesForSourceAndSelector(source, scopeSelector)
+        return this.emitChangeEvent()
+      }
+    } else {
+      for (scopeSelector in this.scopedSettingsStore.propertiesForSource(source)) {
+        this.unset(keyPath, {scopeSelector, source})
+      }
+      if ((keyPath != null) && (source === this.getUserConfigPath())) {
+        return this.set(keyPath, getValueAtKeyPath(this.defaultSettings, keyPath))
+      }
+    }
+  }
+
+  // Extended: Get an {Array} of all of the `source` {String}s with which
+  // settings have been added via {::set}.
+  getSources () {
+    return _.uniq(_.pluck(this.scopedSettingsStore.propertySets, 'source')).sort()
+  }
+
+  // Extended: Retrieve the schema for a specific key path. The schema will tell
+  // you what type the keyPath expects, and other metadata about the config
+  // option.
+  //
+  // * `keyPath` The {String} name of the key.
+  //
+  // Returns an {Object} eg. `{type: 'integer', default: 23, minimum: 1}`.
+  // Returns `null` when the keyPath has no schema specified, but is accessible
+  // from the root schema.
+  getSchema (keyPath) {
+    const keys = splitKeyPath(keyPath)
+    let { schema } = this
+    for (let key of Array.from(keys)) {
```

Comment:
i think `splitKeyPath()` returns an array right? probably don't need this`Array.from()` either.